### PR TITLE
[Feature](function) support split_by_regexp function

### PR DESCRIPTION
**Pull request by zhangstar333**
## Proposed changes
docs link:
https://github.com/apache/doris-website/pull/904

### CODE REVIEW

**Comment by zhangstar333** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/visitor/ScalarFunctionVisitor.java [hunk 1]
```diff
@@ -1828,6 +1829,10 @@ default R visitSplitByString(SplitByString splitByString, C context) {
         return visitScalarFunction(splitByString, context);
     }
 
+    default R visitSplitByRegexp(SplitByRegexp splitByRegexp, C context) {
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/visitor/ScalarFunctionVisitor.java:1832]: please add in lexicographical order

File: regression-test/suites/query_p0/sql_functions/string_functions/test_split_by_regexp.groovy [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    sql """ INSERT INTO ${tableName1} VALUES(10, null, ',') """
+    sql """ INSERT INTO ${tableName1} VALUES(11, 'a,b,c,12345,', ',') """
+
+    test {
+        sql " select split_by_regexp(NULL, 'a12bc23de345f', k1) from test_split_by_regexp"
+        exception "function must be a constant"
+    }
+    qt_select5 "select split_by_regexp(v1, ',') from test_split_by_regexp order by k1;"
+    qt_select6 "select split_by_regexp('do,ris', v2) from test_split_by_regexp order by k1;"
+    qt_select7 "select split_by_regexp(v1, v2) from test_split_by_regexp order by k1;"
```
Comment (morrySnow) [inline regression-test/suites/query_p0/sql_functions/string_functions/test_split_by_regexp.groovy:58]: test the third argument is nagtive value or null

**Comment by zhangstar333** [comment]
run buildall

**Comment by zhangstar333** [comment]
run buildall

**Comment by zhangstar333** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/scalar/SplitByRegexp.java [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+            return new SplitByRegexp(children.get(0), children.get(1));
+        } else {
+            return new SplitByRegexp(children.get(0), children.get(1), children.get(2));
+        }
+    }
+
+    @Override
+    public void checkLegalityBeforeTypeCoercion() {
+        if (children().size() == 3) {
+            if (!child(2).isConstant() || ((Literal) child(2)).getDouble() < 0) {
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/scalar/SplitByRegexp.java:80]: constant is not always literal, test `1 + 2`

**Review by LiBinfeng-01 — APPROVED** [review]

**Review by HappenLee — APPROVED** [review]
LGTM

**Comment by zhangstar333** [comment]
run buildall

**Pull request merged**